shell/platform/android/io/flutter/embedding/engine/FlutterEngine.java

```diff
@@ -287,6 +294,10 @@ public FlutterEngine(
     systemChannel = new SystemChannel(dartExecutor);
     textInputChannel = new TextInputChannel(dartExecutor);
 
+    if (dynamicFeatureManager != null) {
+      dynamicFeatureManager.setDynamicFeatureChannel(dynamicFeatureChannel);
```

Comment:
if you create circular connections, best to break them yourself. Especially when there are singletons like JNI involved.